src/object.rs

```diff
@@ -177,6 +179,51 @@ impl Object {
         Object::from_bytes(&s)
     }
 
+    /// Returns an object from a hash in a git repository.
+    pub fn from_hash(repo: &Repo, hash: &str) -> Result<Object> {
+        let (short_hash, long_hash) = hash.split_at(2);
```

Comment:
Won't this panic if `hash` is less than 2 characters? Maybe `str::split_at_checked` would be nicer?

Reply:
Done. Had to update to 1.80.0 as that function was unstable previously.